### PR TITLE
chore(tracking): Render artsy guarantee component for private artworks + add tap event tracking

### PR DESCRIPTION
**Pull request by jpotts244**
This PR resolves [AMBER-669](https://artsyproduct.atlassian.net/browse/AMBER-669)

### Description

Adds Artsy Guarantee componenet to private artworks screen + add tracking event on Learn more taps

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Display Artsy Guarantee to private artworks screen

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Add analytics tracking event to Artsy Guarantee learn more tap

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[AMBER-669]: https://artsyproduct.atlassian.net/browse/AMBER-669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Artwork/Artwork.tsx [hunk 1]
```diff
@@ -361,6 +361,11 @@ export const Artwork: React.FC<ArtworkProps> = (props) => {
         excludeSeparator: true,
       })
 
+      sections.push({
+        key: "artsyGuarantee",
+        element: <ArtsyGuarantee />,
+      })
+
```
Comment (jpotts244) [inline src/app/Scenes/Artwork/Artwork.tsx:368]: Adds in artsy guarantee section to private artworks experience
Screenshot 2024-05-05 at 4 36 54 PM

Asking Peter to confirm some visuals here: https://artsy.slack.com/archives/C05F8TNKGAV/p1714941889215139

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0aba792ddc7cf1120dc54c96b3a225f86c63f287
</p>

File: src/app/Scenes/Artwork/Artwork.tsx (hunk 1 above)
Comment (damassi) [reply src/app/Scenes/Artwork/Artwork.tsx:368]: 👍 for keeping original component and not creating more 1-offs 🙏

File: src/app/Scenes/Artwork/Components/ArtsyGuarantee.tsx [hunk 2]
```diff
@@ -11,14 +12,30 @@ import {
...
+  const { trackEvent } = useTracking()
+
   const iconProps: IconProps = {
     height: 24,
     width: 24,
     mr: 1,
   }
 
+  const handleLearnMorePress = () => {
+    const payload = {
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ArtsyGuarantee.tsx:27]: For tracking events, we should always assign the type and lock things down (if available): https://github.com/artsy/cohesion/blob/3d1a91515c0092d0f1693453f1deb89e670f9efe/src/Schema/Events/Tap.ts#L925

**Review by damassi — DISMISSED** [review]
One minor change req, otherwise LGTM 👍

File: src/app/Scenes/Artwork/Components/ArtsyGuarantee.tsx (hunk 2 above)
Comment (jpotts244) [reply src/app/Scenes/Artwork/Components/ArtsyGuarantee.tsx:27]: Oops! Good catch 🦅 
Adjusting

Comment (jpotts244) [reply src/app/Scenes/Artwork/Components/ArtsyGuarantee.tsx:27]: Ah I see your point, there are some disgrepencies between the `Click**` events and the `Tap**` events in cohesion

[Tap](https://github.com/artsy/cohesion/blob/3d1a91515c0092d0f1693453f1deb89e670f9efe/src/Schema/Events/Tap.ts#L925-L933) vs [Click](https://github.com/artsy/cohesion/blob/3d1a91515c0092d0f1693453f1deb89e670f9efe/src/Schema/Events/Click.ts#L2103-L2109)

I think louis wanted a `flow` attr for this event
```
context_module:  "artworkDetails"
subject:  "Learn more"
type:  "Link"
flow: "Shipping" | "Artsy Guarantee"
```

Let me ask him if an update to cohesion would be a quick patch

Comment (jpotts244) [reply src/app/Scenes/Artwork/Components/ArtsyGuarantee.tsx:27]: Looking for feedback on adjusting the parent type: https://github.com/artsy/cohesion/pull/498

Comment (damassi) [reply src/app/Scenes/Artwork/Components/ArtsyGuarantee.tsx:27]: Merged 👍

**Review by damassi — APPROVED** [review]

**Pull request merged**